pkg/storage/mvcc_stats_test.go

```diff
@@ -1642,150 +1667,198 @@ func TestMVCCStatsRandomized(t *testing.T) {
+	// Allows easily trimming down the set of actions to
+	// schedule to more easily minimize repro for a known problem.
+	//
+	// When empty, enables all actions (that's what's checked in).
+	enabledActions := map[string]struct{}{}
 
-	actions := make(map[string]func(*state) string)
+	actions := make(map[string]func(*state) (wrote bool, info string))
 
-	actions["Put"] = func(s *state) string {
-		if err := MVCCPut(ctx, s.eng, s.MS, s.key, s.TS, hlc.ClockTimestamp{}, s.rngVal(), s.Txn); err != nil {
-			return err.Error()
+	actions["Put"] = func(s *state) (bool, string) {
+		if err := MVCCPut(ctx, s.batch, s.MSDelta, s.key, s.TS, hlc.ClockTimestamp{}, s.rngVal(), s.Txn); err != nil {
+			return false, err.Error()
 		}
-		return ""
+		return true, ""
 	}
-	actions["InitPut"] = func(s *state) string {
-		failOnTombstones := (s.rng.Intn(2) == 0)
+	actions["InitPut"] = func(s *state) (bool, string) {
+		failOnTombstones := s.rng.Intn(2) == 0
 		desc := fmt.Sprintf("failOnTombstones=%t", failOnTombstones)
-		if err := MVCCInitPut(ctx, s.eng, s.MS, s.key, s.TS, hlc.ClockTimestamp{}, s.rngVal(), failOnTombstones, s.Txn); err != nil {
-			return desc + ": " + err.Error()
+		if err := MVCCInitPut(ctx, s.batch, s.MSDelta, s.key, s.TS, hlc.ClockTimestamp{}, s.rngVal(), failOnTombstones, s.Txn); err != nil {
+			return false, desc + ": " + err.Error()
 		}
-		return desc
+		return true, desc
 	}
-	actions["Del"] = func(s *state) string {
-		if _, err := MVCCDelete(ctx, s.eng, s.MS, s.key, s.TS, hlc.ClockTimestamp{}, s.Txn); err != nil {
-			return err.Error()
+	actions["Del"] = func(s *state) (bool, string) {
+		if _, err := MVCCDelete(ctx, s.batch, s.MSDelta, s.key, s.TS, hlc.ClockTimestamp{}, s.Txn); err != nil {
+			return false, err.Error()
 		}
-		return ""
+		return true, ""
 	}
-	actions["DelRange"] = func(s *state) string {
-		returnKeys := (s.rng.Intn(2) == 0)
-		max := s.rng.Int63n(5)
-		desc := fmt.Sprintf("returnKeys=%t, max=%d", returnKeys, max)
+	actions["DelRange"] = func(s *state) (bool, string) {
 		keyMin := roachpb.KeyMin
 		keyMax := roachpb.KeyMax
 		if s.isLocalKey {
 			keyMax = keys.LocalMax
 		} else {
 			keyMin = keys.LocalMax
 		}
-		if _, _, _, err := MVCCDeleteRange(
-			ctx, s.eng, s.MS, keyMin, keyMax, max, s.TS, hlc.ClockTimestamp{}, s.Txn, returnKeys,
-		); err != nil {
-			return desc + ": " + err.Error()
+
+		mvccRangeDel := s.Txn == nil && s.rng.Intn(2) == 0
+		var mvccRangeDelKey, mvccRangeDelEndKey roachpb.Key
+		if mvccRangeDel {
+			mvccRangeDelKey = keyMin
+			mvccRangeDelEndKey = keyMax
+			n := s.rng.Intn(5)
+			switch n {
+			case 0: // leave as is
+			case 1:
+				mvccRangeDelKey = s.key
+			case 2:
+				mvccRangeDelEndKey = s.key.Next()
+			case 3:
+				mvccRangeDelKey = s.key.Next()
+			case 4:
+				mvccRangeDelEndKey = s.key.Prevish(10)
+			}
 		}
-		return desc
+
+		returnKeys := !mvccRangeDel && (s.rng.Intn(2) == 0)
+		max := s.rng.Int63n(5) // either returnKeys (!mvccRangeDel) or maxIntents.
```

Comment:
nit: Does it really make sense to pass `maxIntents` as anything other than 0 here? It may just error out if there's too many intents.